src/styles/utils/_functions.scss

```diff
@@ -0,0 +1,13 @@
+@each $number, $day in $name_days {
+  .calendar--start-day-#{$day} :first-child {
+    margin-left: calc(#{$number} * #{$field-width});
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--month-length-#{$i} {
+    .calendar__fields:nth-child(#{$i}) {
```

Comment:
It was not working on q&a, because right now you are accessing only one day in from calendar, that is equal to moth length
e.g. .calendar--month-length-30 will only target 30-th child 

Try to come up with solution to select all children after specified day
<details>
  <summary>TIP</summary>
  
use nth-child() with n
  ```
</details>